torch/autograd/grad_mode.py

```diff
@@ -1,8 +1,36 @@
 import torch
 import functools
+import inspect
 
+class _DecoratorContextManager:
+    """Allow a context manager to be used as a decorator"""
 
-class no_grad(object):
+    def __call__(self, func):
+        if inspect.isgeneratorfunction(func):
+            return self._wrap_generator(func)
+
+        @functools.wraps(func)
+        def decorate_no_grad(*args, **kwargs):
```

Comment:
nit: since this is used by both no_grad and enable_grad now, maybe just `decorate_grad`?

Reply:
I've renamed it to `decorate_context` since it would work for any similar context manager.